src/control/run_go_tests.sh

```diff
@@ -56,22 +56,20 @@ if [ "$check" == "false" ]; then
 	setup_environment
 fi
 
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+DIR="$(readlink -f "$(dirname "${BASH_SOURCE[0]}")")"
 
-oldGP=$GOPATH
-export GOPATH=$DIR/../../build/src/control
-echo "GOPATH $GOPATH"
+GOPATH="$(readlink -f "$DIR/../../build/src/control")"
+echo "GOPATH: $GOPATH"
 
 repopath=github.com/daos-stack/daos
+controldir="$GOPATH/src/$repopath/src/control"
 
-# Lint source then run Go tests for each package
-for d in client security server cmd/dmg drpc lib/netdetect; do
-    echo "testing $d"
-    pushd "$GOPATH/src/$repopath/src/control/$d"
-    # todo: provide a sensible way of linting and returning review comments
-    # gofmt -l -s -w . && go tool vet -all . && golint && goimports -w .
-    go test -v
-    popd
-done
+echo "Running all tests under $controldir..."
+pushd "$controldir" >/dev/null
+set +e
+GOPATH="$GOPATH" go test -race -cover -v ./...
```

Comment:
I was following right along until I saw '-race -cover -v ./...'   Could you put a comment in that explains what this line does?

Reply:
From `go help test`:

>         -cover
>             Enable coverage analysis.

From `go help build`:
>         -race
>            enable data race detection.

These are standard build/test flags. We're not doing a whole lot of stuff with goroutines currently, but the race detector should always be enabled for testing because it can catch a lot of subtle issues.